lib/tasks/wikirate.rake

```diff
@@ -48,8 +87,15 @@ namespace :wikirate do
         ActiveRecord::Base.connection.execute 'drop table card_revisions'
         ActiveRecord::Base.connection.execute 'drop table users'
         ActiveRecord::Base.connection.execute 'delete from sessions'
-        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and ca.created_at < '2014'" # delete all webpage++link
-        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and created_at < '2014'"
+        
+        # delete companies
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '651' and id not in ( #{company_ids[0...-1]} )"
+        # delete topics
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '1010' and id not in ( #{topic_ids[0...-1]} )"
+
+        # delete all webpage++link
+        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631','651','699','974','1010','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
+        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631','699','974','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and id not in ( #{card_to_be_kept[0...-1]} )"
         puts "clean database"
```

Comment:
I get it.  I've done much uglier things in desperate times; just hoping we
can get this cleaned up before we forget what it was all about!

On Fri, Mar 13, 2015 at 9:39 AM, xithan <notifications@github.com> wrote:

> In lib/tasks/wikirate.rake
> <https://github.com/wagn/wikirate/pull/118#discussion_r26395882>:
>
> > @@ -48,8 +87,15 @@ namespace :wikirate do
> >          ActiveRecord::Base.connection.execute 'drop table card_revisions'
> >          ActiveRecord::Base.connection.execute 'drop table users'
> >          ActiveRecord::Base.connection.execute 'delete from sessions'
> > -        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and ca.created_at < '2014'" # delete all webpage++link
> > -        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631' '651' '699' '974' '1010' '1109' '1591' '1638' '1690' '2207' '2317' '2327' '2754' '2755' '2813' '2995' '4010' '4030') and created_at < '2014'"
> > +
> > +        # delete companies
> > +        ActiveRecord::Base.connection.execute "delete from cards where type_id = '651' and id not in ( #{company_ids[0...-1]} )"
> > +        # delete topics
> > +        ActiveRecord::Base.connection.execute "delete from cards where type_id = '1010' and id not in ( #{topic_ids[0...-1]} )"
> > +
> > +        # delete all webpage++link
> > +        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631','651','699','974','1010','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and ca.id not in  ( #{card_to_be_kept[0...-1]} )"
> > +        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631','699','974','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and id not in ( #{card_to_be_kept[0...-1]} )"
> >          puts "clean database"
>
> that was part of my desperate trial and error to get the test db from
> 250mb below 100mb. Every test run took about 10 minutes and I didn't
> succeed so the least I cared about was readibility ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wagn/wikirate/pull/118/files#r26395882>.
>



-- 
Ethan McCutchen
One of the Wagneers, Wagn.org

Wagn. How pioneers roll.

s: ethan.mccutchen
t: @intogreater